certora/specs/Vault.spec

```diff
@@ -0,0 +1,990 @@
+    assert getCurvePool(token) == 0;
+
+    env e3;
+    uint16 investPct;
+    setInvestPct(e3, investPct);
+    assert investPct() == investPct;
+
+    env e4;
+    address treasury;
+    setTreasury(e4, treasury);
+    assert treasury() == treasury;
+
+    env e5;
+    uint16 perfFeePct;
+    setPerfFeePct(e5, perfFeePct);
+    assert perfFeePct() == perfFeePct;
+
+    env e6;
+    address s;
+    setStrategy(e6, s);
+    assert strategy() == s;
+
+    env e7;
+    uint16 lossTolerancePct;
+    setLossTolerancePct(e7, lossTolerancePct);
+    assert lossTolerancePct() == lossTolerancePct;
+}
+
+
+/*
+    @Rule
+
+    @Category: variable transition
+
+    @Description:
+        withdrawPerformanceFee function should update state variables correctly and consistently
+*/
+rule integrity_of_withdrawPerformanceFee() {
+    env e;
+    require treasury() != currentContract;
+    require treasury() != strategy();
+
+    underlying.setFee(e, 0);
+
+    uint256 _balanceOfTreasury = underlying.balanceOf(treasury());
+    uint256 _totalUnderlying = totalUnderlying();
+    uint256 _accumulatedPerformanceFee = accumulatedPerfFee();
+
+    withdrawPerformanceFee(e);
+
+    uint256 balanceOfTreasury_ = underlying.balanceOf(treasury());
+    uint256 totalUnderlying_ = totalUnderlying();
+    uint256 accumulatedPerformanceFee_ = accumulatedPerfFee();
+
+    assert balanceOfTreasury_ - _balanceOfTreasury == _accumulatedPerformanceFee;
+    assert _totalUnderlying - totalUnderlying_ == _accumulatedPerformanceFee;
+    assert accumulatedPerformanceFee_ == 0;
+}
+
+
+
+/*
+    @Rule
+
+    @Category: state transition
+
+    @Description:
+        pause and exitPause functions should update states correctly and consistently
+*/
+rule integrity_of_pause_exitPause() {
+    require !paused();
+    env e;
+    pause(e);
+    assert paused();
+
+    env e1;
+    unpause(e1);
+    assert !paused();
+
+    require !exitPaused();
+    env e2;
+    exitPause(e2);
+    assert exitPaused();
+
+    env e3;
+    exitUnpause(e3);
+    assert !exitPaused();
+}
+
+
+/*
+    @Rule
+
+    @Category: unit test
+
+    @Description:
+        yield related calculations are correct
+*/
+rule yield_calculations_correct(address user) {
+    uint256 claimableYield = claimableYield(user);
+    uint256 claimableShares = claimableShares(user);
+    uint256 perfFee = perfFee(user);
+    uint256 yield = claimableYield + perfFee;
+    assert perfFee == pctOf(yield, perfFeePct());
+    assert yield > 0 => claimableShares > 0;
+    assert claimableShares == 0 => yield == 0;
+}
+
+
+/*
+    @Rule
+
+    @Category: unit test
+
+    @Description:
+        maxInvestableAmount equals investPct of totalUnderlying()
+*/
+rule maxInvestableAmount_correct() {
+    assert pctOf(totalUnderlying(), investPct()) == maxInvestableAmount();
+}
+
+
+/*
+    @Rule
+
+    @Category: unit test
+
+    @Description:
+        when the vault is in a loss, withdraw should revert
+*/
+rule withdraw_reverts_when_in_loss() {
```

Comment:
This rule is not saying withdraw ***only*** revert in this condition, it can revert in other conditions as other rules show. but this condition is sufficient to cause withdraw to revert.